src/gui/accelerators.c

```diff
@@ -4207,7 +4207,7 @@ void dt_shortcut_key_press(const dt_input_device_t id,
   }
   else if(g_slist_find_custom(_hold_keys, &this_key, _cmp_key))
   {} // ignore repeating hold key
-  else
+  else if(darktable.view_manager && darktable.view_manager->current_view)
```

Comment:
@ralfbrown @TurboGit 
Instead of fighting symptoms, why not start processing input events only after the gui is up, by moving
```
  g_signal_connect(G_OBJECT(widget), "event",
                   G_CALLBACK(dt_shortcut_dispatcher), NULL);
```
from `dt_gui_gtk_init` to `dt_init`, just after `dt_ctl_switch_mode_to("lighttable");`)

To avoid people assigning `quit` to a midi button and triggering _that_ early, move loading of the shortcuts configs to that same spot as well (from the block just before), so you'd have:
```
  if(init_gui)
  {
    // we have to call dt_ctl_switch_mode_to() here already to not run
    // into a lua deadlock.  having another call later is ok
    dt_ctl_switch_mode_to("lighttable");

    // Save the default shortcuts
    dt_shortcuts_save(".defaults", FALSE);

    // Then load any shortcuts if available (wipe defaults first if requested)
    dt_shortcuts_load(NULL, !dt_conf_get_bool("accel/load_defaults"));

    // Save the shortcuts including defaults
    dt_shortcuts_save(NULL, TRUE);

    // connect the shortcut dispatcher
    g_signal_connect(dt_ui_main_window(darktable.gui->ui), "event", 
                     G_CALLBACK(dt_shortcut_dispatcher), NULL);
```